File: java/insertion_sort.java
public class insertion_sort {
    public static void main(String[] args){
        int[] arr = {8,2,4,1,7,3};
        for (int i=1;i<arr.length;i++){
            for(int j=i;j>0;j--){
                if(arr[j]<arr[j-1]){
                    int temp = arr[j];
                    arr[j]=arr[j-1];
                    arr[j-1]=temp;
                }
                else{
                    break;
                }
            }
        }
        for(int i=0;i<arr.length;i++){
            System.out.println(arr[i]);
        }
    }
    
}

File: python/insertion_sort.py
class insertion_sort:
    def sort(self,arr):
        for i in range(1,len(arr)):
            j=i
            while j>0:
                if arr[j]<arr[j-1]:
                    temp = arr[j]
                    arr[j]=arr[j-1]
                    arr[j-1]=temp
                else:
                    break
                j-=1
        print(arr)

arr=[8,3,5,1,9,2]
a = insertion_sort();
a.sort(arr)